c7n/filters/core.py

```diff
@@ -438,6 +440,35 @@ def validate(self):
                 except re.error as e:
                     raise PolicyValidationError(
                         "Invalid regex: %s %s" % (e, self.data))
+        if 'value_regex' in self.data:
+            return self._validate_value_regex()
+
+        return self
+
+    def _validate_value_regex(self):
+        """ Specific validation for `value_regex` type
+
+        The `value_regex` type works a little differently.
+        In particular it doesn't support OPERATORS that perform operations
+        on a list of values, specifically 'intersect', 'contains', 'difference',
+        'in' and 'not-in'
+        """
+        # Sanity check that we can compile
+        try:
+            pattern = re.compile(self.data['value_regex'])
+            if pattern.groups != 1:
+                raise PolicyValidationError(
+                    "value_regex must have a single capturing group: %s" %
+                    self.data)
+        except re.error as e:
+            raise PolicyValidationError(
+                "Invalid value_regex: %s %s" % (e, self.data))
+
+        if 'op' in self.data:
+            if self.data['op'] in {'intersect', 'in', 'not-in', 'ni', 'contains', 'difference'}:
```

Comment:
@thisisshi is this what your specifically referencing?

Reply:
Yes

Reply:
Consider the behaviour if I have a regex with a non-capturing group e.g. `(?:a|b)([0-9]{4})` and then three resources with tags like `b1111`, `a2222` and `2222`, and a rule that checks if the tag is `in` `['1111', '2222']`. One of them failed the non-capturing group but because it didn't match it will pass through the original value, which will then match the `in` filter anyway, possibly without the user realizing this has happened.

Perhaps it's not a likely scenario, but it seemed more expedient just to protect the user from that sort of thing by not letting them use `in` with this.

Reply:
OK, I've change it to return `None`.

Reply:
Thanks, and given we're returning none explicitly now, i think its safe to omit the additional op restrictions.